src/lib/response-triggers/finish.ts

```diff
@@ -1,26 +1,30 @@
 import Trigger from '../../models/trigger';
-import Winners from './winners';
-import Results from './results';
 import Reset from './reset';
 import Config from '../config';
 import constants from '../../constants';
 import __logger from '../logger';
 import { MessageObject } from 'ciscospark/env';
 import { Role } from '../../models/database';
+import { IPmResultsService } from '../services/pm-results-service';
+import { IResultsService } from '../services/results-service';
+import { IWinnersService } from '../services/winners-service';
 
 const finishCommand = '(?: )*finish(?: )*';
 
 export default class Finish extends Trigger {
-	winners : Winners;
-	results : Results;
+	winnersService: IWinnersService;
+	resultsService: IResultsService;
+	pmResultsService: IPmResultsService;
```

Comment:
Can we not use this horrible Hungarian notation thing from C#?

Reply:
Please suggest a better alternative then. There should be some way to distinguish the interface from the concrete implementation.

Reply:
I mean, personally I find file path more than enough to distinguish. But if you'd really want the name to also be different, my suggestion would be either `PockyDbImpl` or `DefaultPockyDb` for the implementation, while the interface remains `PockyDb`. You've still got a sense of quasi-Hungarian notation which is ugly, but at least then it's hidden from the user most of the time.

Unfortunately I can't find anyone talking about best practice in this regard in Java-/TypeScript. Closest I could find was [this style guide](https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#names) which quite explicitly says it *isn't* meant to be considered a wider community best-practice, but is just the style guide used within the TypeScript development community. There's also [this](https://stackoverflow.com/questions/2814805/java-interfaces-implementation-naming-convention) and [this](https://softwareengineering.stackexchange.com/questions/75956/default-vs-impl-when-implementing-interfaces-in-java), but they're both about Java, not TS, which shares TS's use of "extends" and "interface", rather than C#'s ":" for both, so the extent to which that's applicable is debatable.

Reply:
[Whoops, found this specifically about TypeScript](https://stackoverflow.com/questions/31876947/confused-about-the-interface-and-class-coding-guidelines-for-typescript/41967120#41967120).

Reply:
Ok. I'll agree that we could find a better name for something, but I don't think that calling things WinnerServiceImpl or similar really solves anything

Reply:
I definitely think using `DefaultWinnerService` for the implementation is better than `WinnerServiceImpl`, because it implies that this is the normal default implementation of the `WinnerService` interface, but that there could be others. 

Ideally, if there's something specific *about* the implementation we could use, that would also work. Analogous to the `List` interface being implemented by `ArrayList` and `LinkedList`. Problem is I can't think of what that something would be.

Reply:
yeah, Default prefix is better than Impl, I agree. I was thinking the same about something specific about the implementation, but can't think of anything either

Reply:
Agreed. DefaultWinnersService it is.

Reply:
@jimcullenaus @Lauraducky fixed